### PR TITLE
feat(p2p): add command to show p2p status

### LINKED ISSUE #798: add commands to check p2p connectivity status

### PR DESCRIPTION
**Pull request by srene**
# PR Standards

## Opening a pull request should be able to meet the following requirements

---

Close #798 

<-- Briefly describe the content of this pull request -->

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

File: cmd/dymint/commands/show_p2p_info.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+			peer := p.(map[string]interface{})
+			info := peer["node_info"].(map[string]interface{})
+			status := peer["connection_status"].(map[string]interface{})
+
+			duration, err := strconv.ParseInt(status["Duration"].(string), 10, 64)
+			if err != nil {
+				fmt.Println("Error parsing connection duration:", err)
+			}
+			fmt.Printf("Peer %d Id:%s Multiaddress:%s Duration:%s\n", i, info["id"], peer["remote_ip"], time.Duration(duration))
```
Comment (danwt) [inline cmd/dymint/commands/show_p2p_info.go:105]: what unit is the duration in ?

Comment (danwt) [reply cmd/dymint/commands/show_p2p_info.go:105]: please clarify and comment

**Review by danwt — CHANGES_REQUESTED** [review]

File: cmd/dymint/commands/show_p2p_info.go (hunk 1 above)
Comment (srene) [reply cmd/dymint/commands/show_p2p_info.go]: im not sure what you mean. the units are shown in the duration displayed. i added comments in the code for clarification.
 i also updated the code to order peers by oldest connection and get the id also from rpc.

File: cmd/dymint/commands/show_p2p_info.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+var ip, port string // used for flags
+
+type peerInfo struct {
+	peerId             string
+	multiAddress       string
+	connectionDuration time.Duration
+}
+
+// ShowNodeIDCmd dumps node's ID to the standard output.
+var ShowP2PInfoCmd = &cobra.Command{
```
Comment (mtsitrin) [inline cmd/dymint/commands/show_p2p_info.go:27]: if all it does is calling `rpc.NetInfo` than the implementation is redundant

it should be implemented on the rollapp/RDK similar to 

```
func getNodeStatus(clientCtx client.Context) (*coretypes.ResultStatus, error) {
	node, err := clientCtx.GetNode()
	if err != nil {
		return &coretypes.ResultStatus{}, err
	}

	return node.Status(context.Background())
}
```

but just calling `node.NetInfo`

Comment (srene) [reply cmd/dymint/commands/show_p2p_info.go:27]: got it. implemented in the rdk in https://github.com/dymensionxyz/dymension-rdk/pull/420

**Comment by srene** [comment]
closing pr, since logic has been moved to rdk in https://github.com/dymensionxyz/dymension-rdk/pull/420

**Pull request closed without merging**